client/allocrunner/networking_bridge_linux.go

```diff
@@ -156,69 +165,61 @@ func (b *bridgeNetworkConfigurator) Teardown(ctx context.Context, alloc *structs
 	return b.cni.Teardown(ctx, alloc, spec)
 }
 
-func buildNomadBridgeNetConfig(b bridgeNetworkConfigurator, withConsulCNI bool) []byte {
-	var consulCNI string
-	if withConsulCNI {
-		consulCNI = consulCNIBlock
+func buildNomadBridgeNetConfig(b bridgeNetworkConfigurator, withConsulCNI bool) ([]byte, error) {
+	// Update website/content/docs/networking/cni.mdx when the bridge configuration
+	// is modified. If CNI plugins are added or versions need to be updated for new
+	// fields, add a new constraint to nomad/job_endpoint_hooks.go
+	var ranges []Range
+	allocRange := Range{Subnet: b.allocSubnet}
+	ranges = append(ranges, allocRange)
+
+	var routes []Route
+	allocRoute := Route{Dst: "0.0.0.0/0"}
+	routes = append(routes, allocRoute)
+
+	loop := Loopback{Type: "loopback"}
+
+	NewFirewall := Firewall{
+		Type:                   "firewall",
+		Backend:                "iptables",
+		IptablesAdminChainName: cniAdminChainName,
+	}
+	NewPortmap := Portmap{
+		Type:         "portmap",
+		Capabilities: CapabilityArgs{Portmappings: true},
+		Snat:         true,
+	}
+
+	NewBridge := &Bridge{
+		Type:         "bridge",
+		Bridgename:   b.bridgeName,
+		IpMasq:       true,
+		IsGateway:    true,
+		ForceAddress: true,
+		HairpinMode:  b.hairpinMode,
+		Ipam: IPAM{
+			Type:   "host-local",
+			Routes: routes,
+		},
 	}
+	NewBridge.Ipam.Ranges = append(NewBridge.Ipam.Ranges, ranges)
 
-	return []byte(fmt.Sprintf(nomadCNIConfigTemplate,
-		b.bridgeName,
-		b.hairpinMode,
-		b.allocSubnet,
-		cniAdminChainName,
-		consulCNI,
-	))
-}
+	pluginsList := []any{loop, NewBridge, NewFirewall, NewPortmap}
+	CNIconfig := &BridgeCNIPlugin{
+		CniVersion: "0.4.0",
+		Name:       "nomad",
+		Plugins:    pluginsList,
+	}
+	if withConsulCNI {
+		CNIconfig.Plugins = append(CNIconfig.Plugins, ConsulCNIBlock{
+			Type:     "consul-cni",
+			Loglevel: "debug",
+		})
+	}
 
-// Update website/content/docs/networking/cni.mdx when the bridge configuration
-// is modified. If CNI plugins are added or versions need to be updated for new
-// fields, add a new constraint to nomad/job_endpoint_hooks.go
-const nomadCNIConfigTemplate = `{
```

Comment:
we will want to retain the ability to construct this json for pasting in docs.  I was thinking a unit test that prints out the different variations (and maybe also make some assertions, heh)?  that way the docs could be be kept up to date about as easily as they are now with this template situation.